azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -53,13 +57,39 @@ void restartExecutableFlow(final ExecutableFlow originalExFlow) {
       logger.error(e.getMessage());
       return;
     }
-    final ExecutableFlow retryExFlow =
+    ExecutableFlow retryExFlow =
         this.executorManagerAdapter.createExecutableFlow(project, flow);
+
+    final ExecutionOptions options = originalExFlow.getExecutionOptions();
+
+    final String restartStrategy =
+        options.getFlowParameters().getOrDefault(FLOW_PARAM_RESTART_STRATEGY, "").trim();
+
+    // default strategy - not applying anything on the new one, so as like a new execution
+    if (restartStrategy.isEmpty() || restartStrategy.equals(FLOW_RESTART_STRATEGY_DEFAULT)) {
+      logger.info(String.format("Use default strategy when restarting the execution %s",
+          originalExFlow.getExecutionId()));
+    } else {
+      // non-default strategies
+      if (restartStrategy.equals(FLOW_RESTART_STRATEGY_DISABLE_SUCCEEDED_NODES)){
+        try {
+          disableSucceededSkippedJobsInRetryFlow(originalExFlow, retryExFlow);
+        } catch (ExecutorManagerException e){
+          // rebuild the execution in case data-structure already messed-up
+          retryExFlow = this.executorManagerAdapter.createExecutableFlow(project, flow);
+          logger.warn(String.format(
+              "fail to apply %s restart-strategy for the execution %s, fallback to default method",
+                  FLOW_RESTART_STRATEGY_DISABLE_SUCCEEDED_NODES,
+                  originalExFlow.getExecutionId()),
+              e);
```

Comment:
Wouldn't defaulting to start over strategy be unexpected behavior from user's perspective? They specifically set "start from fail nodes" but instead they notice that Azkaban rerun the entire dag for some reason not visible to them. Why not failing the retry on error and notify users that won't be one for XYZ reason?